agent/acs/handler/task_manifest_responder.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handler
+
+import (
+	"strconv"
+
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/data"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// taskComparer implements the TaskComparer interface defined in ecs-agent module.
+type taskComparer struct {
+	taskEngine engine.TaskEngine
+}
+
+// NewTaskComparer creates a new taskComparer.
+func NewTaskComparer(taskEngine engine.TaskEngine) *taskComparer {
+	return &taskComparer{
+		taskEngine: taskEngine,
+	}
+}
+
+// sequenceNumberAccessor implements the SequenceNumberAccessor interface defined in ecs-agent module.
+type sequenceNumberAccessor struct {
+	latestSeqNumberTaskManifest *int64
+	dataClient                  data.Client
+}
+
+// NewSequenceNumberAccessor creates a new NewSequenceNumberAccessor.
+func NewSequenceNumberAccessor(latestSeqNumberTaskManifest *int64, dataClient data.Client) *sequenceNumberAccessor {
+	return &sequenceNumberAccessor{
+		latestSeqNumberTaskManifest: latestSeqNumberTaskManifest,
+		dataClient:                  dataClient,
+	}
+}
+
+func (tc *taskComparer) CompareRunningTasksOnInstanceWithManifest(
+	message *ecsacs.TaskManifestMessage) ([]*ecsacs.TaskIdentifier, error) {
+	runningTasksOnInstance, err := tc.taskEngine.ListTasks()
```

Comment:
[non-blocking] can we rename this to `allTasksOnInstance` since we are actually filtering tasks in running state on line 63?

Reply:
Good call out! This variable name was pulled in from the existing `agent/acs/handler/task_manifest_handler.go` file, but agreed that something like `allTasksOnInstance` is a more accurate variable name. If other blocking changes need to be made, I'll be sure to rename this (otherwise, I'll rename in a followup PR).